File: Server/src/main/java/com/api/server/foodb/Enzyme.java
package com.api.server.foodb;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "enzymes")
public class Enzyme {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @Size(max = 255)
    @NotNull
    @Column(name = "name", nullable = false)
    private String name;

    @Size(max = 255)
    @Column(name = "gene_name")
    private String geneName;

    @Lob
    @Column(name = "description")
    private String description;

    @Lob
    @Column(name = "go_classification")
    private String goClassification;

    @Lob
    @Column(name = "general_function")
    private String generalFunction;

    @Lob
    @Column(name = "specific_function")
    private String specificFunction;

    @Lob
    @Column(name = "pathway")
    private String pathway;

    @Lob
    @Column(name = "reaction")
    private String reaction;

    @Size(max = 255)
    @Column(name = "cellular_location")
    private String cellularLocation;

    @Lob
    @Column(name = "signals")
    private String signals;

    @Lob
    @Column(name = "transmembrane_regions")
    private String transmembraneRegions;

    @Column(name = "molecular_weight", precision = 15, scale = 9)
    private BigDecimal molecularWeight;

    @Column(name = "theoretical_pi", precision = 15, scale = 9)
    private BigDecimal theoreticalPi;

    @Size(max = 255)
    @Column(name = "locus")
    private String locus;

    @Size(max = 255)
    @Column(name = "chromosome")
    private String chromosome;

    @Size(max = 255)
    @Column(name = "uniprot_name")
    private String uniprotName;

    @Size(max = 100)
    @Column(name = "uniprot_id", length = 100)
    private String uniprotId;

    @Size(max = 10)
    @Column(name = "pdb_id", length = 10)
    private String pdbId;

    @Size(max = 20)
    @Column(name = "genbank_protein_id", length = 20)
    private String genbankProteinId;

    @Size(max = 20)
    @Column(name = "genbank_gene_id", length = 20)
    private String genbankGeneId;

    @Size(max = 20)
    @Column(name = "genecard_id", length = 20)
    private String genecardId;

    @Size(max = 20)
    @Column(name = "genatlas_id", length = 20)
    private String genatlasId;

    @Size(max = 20)
    @Column(name = "hgnc_id", length = 20)
    private String hgncId;

    @Size(max = 255)
    @Column(name = "hprd_id")
    private String hprdId;

    @Size(max = 255)
    @Column(name = "organism")
    private String organism;

    @Lob
    @Column(name = "general_citations")
    private String generalCitations;

    @Lob
    @Column(name = "comments")
    private String comments;

    @Column(name = "creator_id")
    private Integer creatorId;

    @Column(name = "updater_id")
    private Integer updaterId;

    @Column(name = "created_at")
    private Instant createdAt;

    @Column(name = "updated_at")
    private Instant updatedAt;

}

File: DotNetServer/DotNetServer/Core/Entities/Foodb/Enzyme.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace DotNetServer.Core.Entities.Foodb;

[Table("enzymes")]
[Index("GenatlasId", Name = "index_enzymes_on_genatlas_id", IsUnique = true)]
[Index("GenbankGeneId", Name = "index_enzymes_on_genbank_gene_id", IsUnique = true)]
[Index("GenbankProteinId", Name = "index_enzymes_on_genbank_protein_id", IsUnique = true)]
[Index("GeneName", Name = "index_enzymes_on_gene_name", IsUnique = true)]
[Index("GenecardId", Name = "index_enzymes_on_genecard_id", IsUnique = true)]
[Index("HgncId", Name = "index_enzymes_on_hgnc_id", IsUnique = true)]
[Index("HprdId", Name = "index_enzymes_on_hprd_id", IsUnique = true)]
[Index("Name", Name = "index_enzymes_on_name", IsUnique = true)]
[Index("PdbId", Name = "index_enzymes_on_pdb_id", IsUnique = true)]
[Index("UniprotId", Name = "index_enzymes_on_uniprot_id", IsUnique = true)]
[Index("UniprotName", Name = "index_enzymes_on_uniprot_name", IsUnique = true)]
public partial class Enzyme
{
    [Key] [Column("id")] public int Id { get; set; }

    [Column("name")] public string Name { get; set; } = null!;

    [Column("gene_name")] public string? GeneName { get; set; }

    [Column("description", TypeName = "mediumtext")]
    public string? Description { get; set; }

    [Column("go_classification", TypeName = "mediumtext")]
    public string? GoClassification { get; set; }

    [Column("general_function", TypeName = "mediumtext")]
    public string? GeneralFunction { get; set; }

    [Column("specific_function", TypeName = "mediumtext")]
    public string? SpecificFunction { get; set; }

    [Column("pathway", TypeName = "mediumtext")]
    public string? Pathway { get; set; }

    [Column("reaction", TypeName = "mediumtext")]
    public string? Reaction { get; set; }

    [Column("cellular_location")]
    [StringLength(255)]
    public string? CellularLocation { get; set; }

    [Column("signals", TypeName = "mediumtext")]
    public string? Signals { get; set; }

    [Column("transmembrane_regions", TypeName = "mediumtext")]
    public string? TransmembraneRegions { get; set; }

    [Column("molecular_weight")]
    [Precision(15, 9)]
    public decimal? MolecularWeight { get; set; }

    [Column("theoretical_pi")]
    [Precision(15, 9)]
    public decimal? TheoreticalPi { get; set; }

    [Column("locus")] [StringLength(255)] public string? Locus { get; set; }

    [Column("chromosome")]
    [StringLength(255)]
    public string? Chromosome { get; set; }

    [Column("uniprot_name")] public string? UniprotName { get; set; }

    [Column("uniprot_id")]
    [StringLength(100)]
    public string? UniprotId { get; set; }

    [Column("pdb_id")] [StringLength(10)] public string? PdbId { get; set; }

    [Column("genbank_protein_id")]
    [StringLength(20)]
    public string? GenbankProteinId { get; set; }

    [Column("genbank_gene_id")]
    [StringLength(20)]
    public string? GenbankGeneId { get; set; }

    [Column("genecard_id")]
    [StringLength(20)]
    public string? GenecardId { get; set; }

    [Column("genatlas_id")]
    [StringLength(20)]
    public string? GenatlasId { get; set; }

    [Column("hgnc_id")] [StringLength(20)] public string? HgncId { get; set; }

    [Column("hprd_id")] public string? HprdId { get; set; }

    [Column("organism")]
    [StringLength(255)]
    public string? Organism { get; set; }

    [Column("general_citations", TypeName = "mediumtext")]
    public string? GeneralCitations { get; set; }

    [Column("comments", TypeName = "mediumtext")]
    public string? Comments { get; set; }

    [Column("creator_id")] public int? CreatorId { get; set; }

    [Column("updater_id")] public int? UpdaterId { get; set; }

    [Column("created_at", TypeName = "datetime")]
    public DateTime? CreatedAt { get; set; }

    [Column("updated_at", TypeName = "datetime")]
    public DateTime? UpdatedAt { get; set; }
}